prow/plugins/bugzilla/bugzilla.go

```diff
@@ -356,12 +356,13 @@ To reference a bug, add 'Bug XXX:' to the title of this pull request and request
 			log.Debug("Valid bug found.")
 			response = fmt.Sprintf(`This pull request references `+bugLink+`, which is valid.`, e.bugId, bc.Endpoint(), e.bugId)
 			// if configured, move the bug to the new state
-			if options.StatusAfterValidation != nil && bug.Status != *options.StatusAfterValidation {
-				if err := bc.UpdateBug(e.bugId, bugzilla.BugUpdate{Status: *options.StatusAfterValidation}); err != nil {
+			update := options.StateAfterValidation.AsBugUpdate()
+			if update != nil && (update.Status != bug.Status || update.Resolution != bug.Resolution) {
```

Comment:
Why not a `.Equal()`? I imagine we will do this type of comparison a bunch

Reply:
Thanks, a refactor artifact